viewer/Viewer.cpp

```diff
@@ -784,62 +989,141 @@ namespace Enki
 		updateGL();
 	}
 	
+	void ViewerWidget::keyPressEvent(QKeyEvent* event)
+	{
+		if (event->key() == Qt::Key_F1)
+			showHelp();
+	}
+
 	void ViewerWidget::mousePressEvent(QMouseEvent *event)
 	{
+		// initialization
 		mouseGrabPos = event->pos();
-		/*if (event->button() == Qt::RightButton)
+
+		// change selected object
+		if (event->buttons() & Qt::LeftButton)
 		{
-			mouseGrabbed = true;
-			mouseGrabPos = event->pos();
-		}*/
+			if(selectedObject != pointedObject) trackballView = false;
+			selectedObject = pointedObject;
+		}
+
+		// code button identification for robot
+		int buttonCode = 0;
```

Comment:
This should be a bit mask, as several mouse buttons can be pressed at the same time. Also, it should be an `unsigned`.